source/runtime.cpp

```diff
@@ -247,10 +247,16 @@ void reshade::runtime::load_effect(const std::filesystem::path &path, size_t &ou
 		}
 
 		// Pre-process the source file
-		if (!pp.append_file(path))
+		if (pp.append_file(path))
+		{
+			effect.user_definitions = std::move(pp.user_definitions());
+		}
+		else
 		{
 			LOG(ERROR) << "Failed to load " << path << ":\n" << pp.errors();
 			effect.compile_sucess = false;
+
+			effect.user_definitions.clear();
```

Comment:
This is not necessary. `effect` is a new object, so `user_definitions` will always be empty here anyway. There is nothing to clear. Either way, doesn't matter much, just do this for simplicity:
```c++
if (!pp.append_file(path))
{
...
}

effect.user_definitions = std::move(pp.user_definitions());
```